alexandria/core/tasks.py

```diff
@@ -0,0 +1,55 @@
+import hashlib
+from pathlib import Path
+
+import tika.language
+import tika.parser
+from django.conf import settings
+from django.contrib.postgres.search import SearchVector
+from django.db.models import Value
+
+from alexandria.core.models import File
+from celery import shared_task
+
+
+@shared_task(soft_time_limit=301)
+def set_content_vector(file_pk: str):
+    file = File.objects.get(pk=file_pk)
+    file.content.file.file.seek(0)
+
+    # tika has an internal time limit of 300s, set the request limit to match that
+    # different values should be set in tika as well
+    # https://github.com/CogStack/tika-service/blob/master/README.md#tika-parsers-configuration
+    parsed_content = tika.parser.from_buffer(
+        file.content.file.file, requestOptions={"timeout": 300}
+    )
+
+    name_vector = SearchVector(Value(Path(file.name).stem), weight="A")
+    if not parsed_content["content"]:
+        file.content_vector = name_vector
+        file.save()
```

Comment:
```suggestion
        # Update only content vector, to avoid race with the checksum task
        File.objects.filter(pk=file.pk).update(content_vector=file.content_vector)
```